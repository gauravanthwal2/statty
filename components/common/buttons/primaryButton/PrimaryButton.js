import styles from "./PrimaryButton.module.css";

const PrimaryButton = ({ text, ...buttoProps }) => {
  return (
    <button
      className={styles.primaryButton}
      {...buttoProps}
      data-testid="primary"
    >
      {text}
    </button>
  );
};

export default PrimaryButton;
