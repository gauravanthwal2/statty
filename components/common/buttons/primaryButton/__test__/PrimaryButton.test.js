import userEvent from "@testing-library/user-event";
import {
  renderWithProvider,
  cleanup,
  fireEvent,
} from "../../../../../utils/test-utils";
import PrimaryButton from "../PrimaryButton";

const AddParticipantButtonText = "Add Participant";
const UpdateParticipantButtonText = "Update Participant";
const AddTrainingButtonText = "Add New Training";
const UpdateTrainingButtonText = "Update Training";

describe("button renders", () => {
  afterEach(() => cleanup());
  test("testing primary button on Add training", () => {
    const handleSubmit = jest.fn();
    const { getByRole } = renderWithProvider(
      <PrimaryButton text={AddTrainingButtonText} onClick={handleSubmit} />
    );
    const button = getByRole("button");
    expect(button).toHaveTextContent(AddTrainingButtonText);
    fireEvent.click(button);
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });

  test("testing primary button on update training", () => {
    const handleSubmit = jest.fn();
    const { getByRole } = renderWithProvider(
      <PrimaryButton text={UpdateTrainingButtonText} onClick={handleSubmit} />
    );
    const button = getByRole("button");
    expect(button).toHaveTextContent(UpdateTrainingButtonText);
    fireEvent.click(button);
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });

  test("testing primary button on add participant", () => {
    const handleSubmit = jest.fn();
    const { getByRole } = renderWithProvider(
      <PrimaryButton text={AddParticipantButtonText} onClick={handleSubmit} />
    );
    const button = getByRole("button");
    expect(button).toHaveTextContent(AddParticipantButtonText);
    fireEvent.click(button);
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });

  test("testing primary button on update participant", () => {
    const handleSubmit = jest.fn();
    const { getByRole } = renderWithProvider(
      <PrimaryButton
        text={UpdateParticipantButtonText}
        onClick={handleSubmit}
      />
    );
    const button = getByRole("button");
    expect(button).toHaveTextContent(UpdateParticipantButtonText);
    fireEvent.click(button);
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });

  test("check if button has a css class", () => {
    const handleSubmit = jest.fn();
    const { getByRole } = renderWithProvider(
      <PrimaryButton
        text={UpdateParticipantButtonText}
        onClick={handleSubmit}
      />
    );
    const button = getByRole("button");
    expect(button).toHaveClass("primaryButton");
  });

  test("renders secondary button  unchanged", () => {
    const handleSubmit = jest.fn();
    const { container } = renderWithProvider(
      <PrimaryButton text={AddParticipantButtonText} onClick={handleSubmit} />
    );
    expect(container).toMatchSnapshot();
  });

  test("check  button attributes", () => {
    const handleSubmit = jest.fn();
    const props = {
      type: "submit",
    };
    const { getByRole } = renderWithProvider(
      <PrimaryButton
        text={AddTrainingButtonText}
        onClick={handleSubmit}
        {...props}
      />
    );
    const button = getByRole("button");
    expect(button).not.toHaveAttribute("disabled");
    expect(button).toHaveAttribute("type", "submit");
    expect(button).not.toHaveAttribute("type", "button");
  });
});
