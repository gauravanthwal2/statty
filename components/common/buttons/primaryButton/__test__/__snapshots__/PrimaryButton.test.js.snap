// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders renders secondary button  unchanged 1`] = `
<div>
  <button
    class="primaryButton"
    data-testid="primary"
  >
    Add Participant
  </button>
</div>
`;
